core/field_angle.js

```diff
@@ -333,6 +341,38 @@ Blockly.FieldAngle.prototype.updateGraph_ = function() {
   this.line_.setAttribute('y2', y2);
 };
 
+/**
+ * Handle key down to the editor.
+ * @param {!Event} e Keyboard event.
+ * @protected
+ * @override
+ */
+Blockly.FieldAngle.prototype.onHtmlInputKeyDown_ = function(e) {
+  Blockly.FieldAngle.superClass_.onHtmlInputKeyDown_.call(this, e);
+
+  var multiplier;
+  if (e.keyCode === Blockly.utils.KeyCodes.LEFT) {
+    // decrement (increment in RTL)
+    multiplier = this.sourceBlock_.RTL ? 1 : -1;
+  } else if (e.keyCode === Blockly.utils.KeyCodes.RIGHT ||
+      e.keyCode === Blockly.utils.KeyCodes.UP) {
+    // increment (decrement in RTL)
+    multiplier = this.sourceBlock_.RTL ? -1 : 1;
+  } else if (e.keyCode === Blockly.utils.KeyCodes.DOWN) {
+    // decrement
+    multiplier = -1;
+  } else if (e.keyCode === Blockly.utils.KeyCodes.UP) {
+    // increment
+    multiplier = 1;
+  }
+  if (multiplier) {
+    this.setAngle(Number(this.getValue()) +
```

Comment:
This is passing in -15 when wrapping instead of 345. I think we need to handle that either here or in setAngle

Reply:
Yep, i'll add wrapping for negative values.